portal/src/graphql/portal/DesignScreen/Components.tsx

```diff
@@ -0,0 +1,559 @@
+  props
+) {
+  const { value, onChange } = props;
+  const { renderToString } = useContext(MFContext);
+  const options = useMemo(
+    () => AllBorderRadiusStyleTypes.map((value) => ({ value })),
+    []
+  );
+
+  const [radiusValue, setRadiusValue] = useState(() => {
+    if (value.type !== "rounded") {
+      return "";
+    }
+    return value.radius;
+  });
+
+  useEffect(() => {
+    if (value.type !== "rounded") {
+      setRadiusValue("");
+    } else {
+      setRadiusValue(value.radius);
+    }
+  }, [value, radiusValue]);
+
+  const onSelectOption = useCallback(
+    (option: Option<BorderRadiusStyleType>) => {
+      if (option.value === "rounded") {
+        onChange({
+          type: option.value,
+          radius: "0",
+        });
+      } else {
+        onChange({
+          type: option.value,
+        });
+      }
+    },
+    [onChange]
+  );
+
+  const onBorderRadiusChange = useCallback(
+    (_: any, value?: string) => {
+      if (value == null) {
+        return;
+      }
+      onChange({
+        type: "rounded",
+        radius: value,
+      });
+    },
+    [onChange]
+  );
+
+  const renderOption = useCallback(
+    (option: Option<BorderRadiusStyleType>, selected: boolean) => {
+      return (
+        <span
+          className={cn(
+            styles.icAlignment,
+            (() => {
+              switch (option.value) {
+                case "none":
+                  return styles.icBorderRadiusSquare;
+                case "rounded":
+                  return styles.icBorderRadiusRounded;
+                case "rounded-full":
+                  return styles.icBorderRadiusFullRounded;
+                default:
+                  return undefined;
+              }
+            })(),
+            selected && styles.selected
+          )}
+        ></span>
+      );
+    },
+    []
+  );
+
+  return (
+    <div>
+      <ButtonToggleGroup
+        value={value.type}
+        options={options}
+        onSelectOption={onSelectOption}
+        renderOption={renderOption}
+      ></ButtonToggleGroup>
+      {value.type === "rounded" ? (
+        <TextField
+          className={cn("mt-3")}
+          label={renderToString(
+            "DesignScreen.configuration.borderRadius.label"
+          )}
+          value={radiusValue}
+          onChange={onBorderRadiusChange}
+        />
+      ) : null}
+    </div>
+  );
+};
+
+type ImageFileExtension = ".jpeg" | ".png" | ".gif";
+function mediaTypeToExtension(mime: string): ImageFileExtension {
+  switch (mime) {
+    case "image/png":
+      return ".png";
+    case "image/jpeg":
+      return ".jpeg";
+    case "image/gif":
+      return ".gif";
+    default:
+      throw new Error(`unsupported media type: ${mime}`);
+  }
+}
+
+interface ImagePickerProps {
+  base64EncodedData: string | null;
+  onChange: (
+    image: {
+      base64EncodedData: string;
+      extension: ImageFileExtension;
+    } | null
+  ) => void;
```

Comment:
Looks like these are duplicated from ImageFilePicker, but these should not be duplicated because they should support same file types.